concordium-node/src/p2p/peers.rs

```diff
@@ -65,14 +66,22 @@ impl P2PNode {
         self.stats.set_bytes_received(bytes_received);
         self.stats.set_bytes_sent(bytes_sent);
 
-        let time_diff = self.config.housekeeping_interval as f64;
-
-        let avg_bps_in =
-            ((bytes_received.saturating_sub(prev_bytes_received)) as f64 / time_diff) as u64;
-        let avg_bps_out = ((bytes_sent.saturating_sub(prev_bytes_sent)) as f64 / time_diff) as u64;
-
+        let now = Utc::now().timestamp_millis(); // Timestamps are stored in millis in order to allow measurements within the
+                                                 // second. Hence the number of bytes received/send must be multiplied by a
+                                                 // factor 1000 in order to obtain bps.
+        let (avg_bpms_in, avg_bpms_out) = calculate_average_throughput(
+            self.stats.get_last_throughput_measurement(),
+            now,
+            prev_bytes_received * 1000,
+            bytes_received * 1000,
+            prev_bytes_sent * 1000,
+            bytes_sent * 1000,
+        );
+
+        let (avg_bps_in, avg_bps_out) = (avg_bpms_in, avg_bpms_out);
```

Comment:
I don't understand what this renaming is for. Shouldn't there be some rescaling?